configure.ac

```diff
@@ -0,0 +1,1657 @@
+    [arm*-*-linux*], [natdynlink=true],
+    [arm*-*-freebsd*], [natdynlink=true],
+    [earm*-*-netbsd*], [natdynlink=true],
+    [aarch64-*-linux*], [natdynlink=true])])
+
+# Try to work around the Skylake/Kaby Lake processor bug.
+AS_CASE(["$CC,$host"],
+  [*gcc*,x86_64-*|*gcc*,i686-*], 
+    [OCAML_CC_HAS_FNO_TREE_VRP
+    AS_IF([$cc_has_fno_tree_vrp],
+      [internal_cflags="$internal_cflags -fno-tree-vrp"])])
+
+# Configure the native-code compiler
+
+arch=none
+model=default
+system=unknown
+
+AS_CASE([$host],
+  [[i[3456]86-*-linux*]],
+    [arch=i386; system=linux_elf],
+  [[i[3456]86-*-*bsd*]],
+    [arch=i386; system=bsd_elf],
+  [[i[3456]86-*-haiku*]],
+    [arch=i386; system=beos],
+  [[i[3456]86-*-cygwin]],
+    [arch=i386; system=cygwin],
+  [[i[3456]86-*-darwin*]],
+    [AS_IF([$arch64], [arch=amd64], [arch=i386]); system=macosx],
+  [[i[3456]86-*-gnu*]],
+    [arch=i386; system=gnu],
+  [[i[3456]86-*-mingw32]],
+    [arch=i386; system=mingw],
+  [i686-pc-windows],
+    [arch=i386; system=win32],
+  [x86_64-pc-windows],
+    [arch=amd64; system=win64],
+  [[powerpc64le*-*-linux*]],
+    [arch=power; model=ppc64le; system=elf],
+  [[powerpc*-*-linux*]],
+    [arch=power; AS_IF([$arch64],[model=ppc64],[model=ppc]); system=elf],
+  [[powerpc-*-netbsd*]],
+    [arch=power; model=ppc; system=netbsd],
+  [[powerpc-*-openbsd*]],
+    [arch=power; model=ppc; system=bsd_elf],
+  [[s390x*-*-linux*]],
+    [arch=s390x; model=z10; system=elf],
+  [armv6*-*-linux-gnueabihf],
+    [arch=arm; model=armv6; system=linux_eabihf],
+  [armv7*-*-linux-gnueabihf],
+    [arch=arm; model=armv7; system=linux_eabihf],
+  [armv8*-*-linux-gnueabihf],
+    [arch=arm; model=armv8; system=linux_eabihf],
+  [armv8*-*-linux-gnueabi],
+    [arch=arm; model=armv8; system=linux_eabi],
+  [armv7*-*-linux-gnueabi],
+    [arch=arm; model=armv7; system=linux_eabi],
+  [armv6t2*-*-linux-gnueabi],
+    [arch=arm; model=armv6t2; system=linux_eabi],
+  [armv6*-*-linux-gnueabi],
+    [arch=arm; model=armv6; system=linux_eabi],
+  [armv6*-*-freebsd*],
+    [arch=arm; model=armv6; system=freebsd],
+  [earmv6*-*-netbsd*],
+    [arch=arm; model=armv6; system=netbsd],
+  [earmv7*-*-netbsd*],
+    [arch=arm; model=armv7; system=netbsd],
+  [armv5te*-*-linux-gnueabi],
+    [arch=arm; model=armv5te; system=linux_eabi],
+  [armv5*-*-linux-gnueabi],
+    [arch=arm; model=armv5; system=linux_eabi],
+  [arm*-*-linux-gnueabi],
+    [arch=arm; system=linux_eabi],
+  [arm*-*-openbsd*],
+    [arch=arm; system=bsd],
+  [zaurus*-*-openbsd*],
+    [arch=arm; system=bsd],
+  [x86_64-*-linux*],
+    [arch=amd64; system=linux],
+  [x86_64-*-gnu*],
+    [arch=amd64; system=gnu],
+  [x86_64-*-dragonfly*],
+    [arch=amd64; system=dragonfly],
+  [x86_64-*-freebsd*],
+    [arch=amd64; system=freebsd],
+  [x86_64-*-netbsd*],
+    [arch=amd64; system=netbsd],
+  [x86_64-*-openbsd*],
+    [arch=amd64; system=openbsd],
+  [x86_64-*-darwin*],
+    [arch=amd64; system=macosx],
+  [x86_64-*-mingw32],
```

Comment:
Same question again with mingw32/mingw*
It seems more important here, since you're turning mingw32 into mingw64.